File: secao09/exploracao.R
# Exploração dos dados

dados = read.csv("data/raw/Churn.csv", sep = ";", na.strings = "", stringsAsFactors = T)
head(dados)
summary(dados)

# Colunas
# Dar nomes às colunas
colnames(dados) <- c("Id", "Score", "Estado", "Genero", "Idade", "Patrimônio", 
                     "Saldo", "Produtos", "TemCartCredito", "Ativo", "Salario", 
                     "Saiu")

head(dados)

# Explorar dados categóricos
## Estados
counts <- table(dados$Estado)
barplot(counts, main="Estado", xlab = "Estado")

## Gênero
counts <- table(dados$Genero)
barplot(counts, main="Gêneros", xlab = "Gêneros")

# Explorar dados numéricos
## Score
summary(dados$Score)
boxplot(dados$Score)
hist(dados$Score)

## Idade
summary(dados$Idade)
boxplot(dados$Idade)
hist(dados$Idade)

## Saldo
summary(dados$Saldo)
boxplot(dados$Saldo)
hist(dados$Saldo)

# Salário
summary(dados$Salario)
boxplot(dados$Salario)
hist(dados$Salario)

File: secao09/exploracao.py
import pandas as pd
pd.set_option("display.max_columns", None)
import seaborn as sns
import statistics as sts

dataset = pd.read_csv('data/raw/Churn.csv', sep=";")
print("Conhecendo a base de dados")
print(dataset.head())

print("Tamanho...")
print(dataset.shape)

# Dar nomes às colunas 
print('Renomeando colunas...')
dataset.columns = ["Id", "Score", "Estado", "Genero", "Idade", "Patrimônio", 
                     "Saldo", "Produtos", "TemCartCredito", "Ativo", "Salario", 
                     "Saiu"]

print(dataset.head())

# ---- Explorar dados categóricos
# Estados
print("Agrupando...")
agrupado = dataset.groupby(['Estado']).size()
print(agrupado)
